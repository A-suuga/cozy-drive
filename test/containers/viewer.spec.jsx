'use strict'

/* eslint-env jest */

import React from 'react'
import { shallow } from 'enzyme'

import Connected, { Viewer } from '../../src/containers/Viewer'

import thunk from 'redux-thunk'
import configureMockStore from 'redux-mock-store'
const middlewares = [ thunk ]
const mockStore = configureMockStore(middlewares)

const routerObjectMock = {
  location: {
    pathname: '/mock/33dda00f0eec15bc3b3c59a615001ac8'
  },
  push: jest.fn()
}

const paramsMock = {
  photoId: '33dda00f0eec15bc3b3c59a615001ac8'
}

const paramsMock2 = {
  photoId: '33dda00f0eec15bc3b3c59a615001ac9'
}

const paramsMock3 = {
  photoId: '33dda00f0eec15bc3b3c59a615001ad0'
}

const photosMock = [
  {
    _id: '33dda00f0eec15bc3b3c59a615001ac8'
  },
  {
    _id: '33dda00f0eec15bc3b3c59a615001ac9'
  },
  {
    _id: '33dda00f0eec15bc3b3c59a615001ad0'
  }
]

const storeMock = mockStore({
  photos: photosMock
})

describe('Viewer component', () => {
  it('should be displayed correctly with the img src computed from photoId using the store and the connected component (current photo is the first of the list)', () => {
    const component = shallow(
      <Connected params={paramsMock} store={storeMock} />
    )
    expect(component.node.props.current).toBe(photosMock[0]._id)
    expect(component.node.props.previous).toBe(photosMock[2]._id)
    expect(component.node.props.next).toBe(photosMock[1]._id)
    expect(component.node).toMatchSnapshot()
  })

  it('should be displayed correctly with the img src computed from photoId using the store and the connected component (current photo is the middle of the list)', () => {
    const component = shallow(
      <Connected params={paramsMock2} store={storeMock} />
    )
    expect(component.node.props.current).toBe(photosMock[1]._id)
    expect(component.node.props.previous).toBe(photosMock[0]._id)
    expect(component.node.props.next).toBe(photosMock[2]._id)
    expect(component.node).toMatchSnapshot()
  })

  it('should be displayed correctly with the img src computed from photoId using the store and the connected component (current photo is the last of the list)', () => {
    const component = shallow(
      <Connected params={paramsMock3} store={storeMock} />
    )
    expect(component.node.props.current).toBe(photosMock[2]._id)
    expect(component.node.props.previous).toBe(photosMock[1]._id)
    expect(component.node.props.next).toBe(photosMock[0]._id)
    expect(component.node).toMatchSnapshot()
  })

  it('should handle correctly onClick events for previous pictures', () => {
    const component = shallow(
      <Viewer
        current={paramsMock.photoId}
        next={1}
        previous={0}
        router={routerObjectMock}
      />
    )
    component.find('a').at(0).simulate('click')
    expect(component.node).toMatchSnapshot()
  })

  it('should handle correctly onClick events for next pictures', () => {
    const component = shallow(
      <Viewer
        current={paramsMock.photoId}
        next={photosMock[1]}
        previous={photosMock[0]}
        router={routerObjectMock}
      />
    )
    component.find('a').at(1).simulate('click')
    expect(component.node).toMatchSnapshot()
  })
})
