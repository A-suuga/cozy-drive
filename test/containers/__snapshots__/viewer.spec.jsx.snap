exports[`Viewer component should be displayed correctly with the img src computed from photoId using the store and the connected component (current photo is the first of the list) 1`] = `
<withRouter(Viewer)
  current="33dda00f0eec15bc3b3c59a615001ac8"
  dispatch={[Function]}
  next="33dda00f0eec15bc3b3c59a615001ac9"
  params={
    Object {
      "photoId": "33dda00f0eec15bc3b3c59a615001ac8",
    }
  }
  previous="33dda00f0eec15bc3b3c59a615001ad0"
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  } />
`;

exports[`Viewer component should be displayed correctly with the img src computed from photoId using the store and the connected component (current photo is the last of the list) 1`] = `
<withRouter(Viewer)
  current="33dda00f0eec15bc3b3c59a615001ad0"
  dispatch={[Function]}
  next="33dda00f0eec15bc3b3c59a615001ac8"
  params={
    Object {
      "photoId": "33dda00f0eec15bc3b3c59a615001ad0",
    }
  }
  previous="33dda00f0eec15bc3b3c59a615001ac9"
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  } />
`;

exports[`Viewer component should be displayed correctly with the img src computed from photoId using the store and the connected component (current photo is the middle of the list) 1`] = `
<withRouter(Viewer)
  current="33dda00f0eec15bc3b3c59a615001ac9"
  dispatch={[Function]}
  next="33dda00f0eec15bc3b3c59a615001ad0"
  params={
    Object {
      "photoId": "33dda00f0eec15bc3b3c59a615001ac9",
    }
  }
  previous="33dda00f0eec15bc3b3c59a615001ac8"
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  } />
`;

exports[`Viewer component should handle correctly onClick events for next pictures 1`] = `
<div
  className="pho-viewer-wrapper"
  role="viewer">
  <_translate />
  <div
    className="pho-viewer-content">
    <a
      className="pho-viewer-nav-previous"
      onClick={[Function]}
      role="button" />
    <div
      className="pho-viewer-photo">
      <img
        src="http://cozy.local:8080/files/download/33dda00f0eec15bc3b3c59a615001ac8" />
    </div>
    <a
      className="pho-viewer-nav-next"
      onClick={[Function]}
      role="button" />
  </div>
</div>
`;

exports[`Viewer component should handle correctly onClick events for previous pictures 1`] = `
<div
  className="pho-viewer-wrapper"
  role="viewer">
  <_translate />
  <div
    className="pho-viewer-content">
    <a
      className="pho-viewer-nav-previous"
      onClick={[Function]}
      role="button" />
    <div
      className="pho-viewer-photo">
      <img
        src="http://cozy.local:8080/files/download/33dda00f0eec15bc3b3c59a615001ac8" />
    </div>
    <a
      className="pho-viewer-nav-next"
      onClick={[Function]}
      role="button" />
  </div>
</div>
`;
